platform/representation/src/gestures/DropGesture.js

```diff
@@ -90,25 +107,61 @@ define(
                 }
             }
 
+            function createVirtualPanel(base, overlayId){
+
+                var typeKey = 'telemetry.panel',
+                    type = typeService.getType(typeKey),
+                    model = type.getInitialModel(),
+                    id = uuid(),
+                    newPanel = undefined;
+
+                model.type = typeKey;
+                model.name = 'New telemetry panel';
+                newPanel = instantiate(model, id);
+
+                [base.getId(), overlayId].forEach(function(id){
+                    newPanel.getCapability('composition').add(id)
+                });
+
+                newPanel.getCapability('location').setPrimaryLocation(base.getCapability('location').getContextualLocation());
+
+                //ObjectService is wrapped by a decorator which is obscuring
+                // the newObject method.
+                var virtualPanel = new EditableDomainObject(newPanel, $q);
+                virtualPanel.setOriginalObject(base);
+                return virtualPanel;
+
+            }
+
             function drop(e) {
                 var event = (e || {}).originalEvent || e,
                     id = event.dataTransfer.getData(GestureConstants.MCT_DRAG_TYPE),
-                    domainObjectType = domainObject.getModel().type;
-                
+                    domainObjectType = editableDomainObject.getModel().type,
+                    virtualObj;
+
                 // If currently in edit mode allow drag and drop gestures to the
                 // domain object. An exception to this is folders which have drop
                 // gestures in browse mode.
-                if (domainObjectType === 'folder' || domainObject.hasCapability('editor')) {
+                //if (domainObjectType === 'folder' || domainObject.hasCapability('editor')) {
                 
                     // Handle the drop; add the dropped identifier to the
                     // destination domain object's composition, and persist
                     // the change.
                     if (id) {
-                        $q.when(action && action.perform()).then(function (result) {
```

Comment:
I think it makes sense that `compose` (along with `move`, `copy`, `link`) should return the relevant domain object (or a promise for it) in its new context.

The question then becomes - who is responsible for navigating to that object? I see a few alternatives:

* Broadcast out creation on a topic; listen to this topic in Browse and navigate as objects are created.
  * Con: This would probably make some stuff happen
* Just include the navigation change in the `create` action (and in whatever action makes the new virtual panel)
* Decorate the `create` actions with this behavior (perhaps from Browse mode's create button specifically, if we really want to confine the scope of this behavior). This means doing similar from the virtual panel.
  * Perhaps the virtual panel can itself be handled with a `create` action for a telemetry panel? Would need to do something about the persistence invocations in create (maybe pass a flag in action context to suppress this)